### PR TITLE
solved task 5 months with days

### PR DESCRIPTION
**Pull request by neb5562**

### CODE REVIEW

File: 5 - months with days/solution_5.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'date'
+NEGATIVE_NUM = -1
+
+class Dote
```
Comment (alexshvedko) [inline 5 - months with days/solution_5.rb]: Dote? :)

File: 5 - months with days/solution_5.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  def initialize(year)
+    @year = year
+    @months_with_days = generate_months_with_days_count
+  end
+
+  def generate_months_with_days_count
+    months_with_days = {}
+    (1..12).each do |i|
```
Comment (alexshvedko) [inline 5 - months with days/solution_5.rb]: magic numbers. For example: `beggining_of_month`

File: 5 - months with days/solution_5.rb (hunk 1 above)
Comment (neb5562) [reply 5 - months with days/solution_5.rb]: > Dote? :)

Changed to `NewDate`

File: 5 - months with days/solution_5.rb [hunk 3]
```diff
@@ -9,7 +11,7 @@ def initialize(year)
 
   def generate_months_with_days_count
     months_with_days = {}
-    (1..12).each do |i|
+    (FIRST_MONTH..LAST_MONTH).each do |i|
```
Comment (neb5562) [inline 5 - months with days/solution_5.rb:20]: @alexshvedko fixed magic numbers. declared two constants. `FIRST_MONTH` and `LAST_MONTH`

File: 5 - months with days/solution_5.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    months_with_days = {}
+    (1..12).each do |i|
+      months_with_days[Date::MONTHNAMES[i].to_s] = Date.new(@year, i, NEGATIVE_NUM).day
+    end
+    months_with_days
+  end
+
+  # select months of year with 30 days or any value days variable gets.
+  def get_months_with_days days
+    @months_with_days.select{ |month_name, days_in_month| days_in_month == days }
```
Comment (alexshvedko) [inline 5 - months with days/solution_5.rb]: you defined the variable `month_name`, but don't use it. in this case, better defined as `_`
`  @months_with_days.select{ |_, days_in_month| days_in_month == days }`

File: 5 - months with days/solution_5.rb [hunk 5]
```diff
@@ -19,7 +19,7 @@ def generate_months_with_days_count
 
   # select months of year with 30 days or any value days variable gets.
   def get_months_with_days days
-    @months_with_days.select{ |month_name, days_in_month| days_in_month == days }
+    @months_with_days.select{ |_, days_in_month| days_in_month == days }
```
Comment (neb5562) [inline 5 - months with days/solution_5.rb]: fixed @alexshvedko

File: 5 - months with days/solution_5.rb (hunk 2 above)
Comment (neb5562) [reply 5 - months with days/solution_5.rb]: @alexshvedko fixed

File: 5 - months with days/solution_5.rb (hunk 4 above)
Comment (neb5562) [reply 5 - months with days/solution_5.rb]: @alexshvedko done

**Review by alexshvedko — APPROVED** [review]

File: 5 - months with days/solution_5.rb [hunk 6]
```diff
@@ -0,0 +1,27 @@
+require 'date'
+NEGATIVE_NUM = -1
```
Comment (cyberik-dev) [inline 5 - months with days/solution_5.rb]: Please move constants to the class

File: 5 - months with days/solution_5.rb [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+  def generate_months_with_days_count
+    months_with_days = {}
+    (FIRST_MONTH..LAST_MONTH).each do |i|
+      months_with_days[Date::MONTHNAMES[i].to_s] = Date.new(@year, i, NEGATIVE_NUM).day
+    end
+    months_with_days
+  end
+
+  # select months of year with 30 days or any value days variable gets.
+  def get_months_with_days days
```
Comment (cyberik-dev) [inline 5 - months with days/solution_5.rb]: https://github.com/rubocop/ruby-style-guide#method-definition-parentheses

**Review by cyberik-dev — CHANGES_REQUESTED** [review]

File: 5 - months with days/solution_5.rb [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+    @months_with_days.select{ |_, days_in_month| days_in_month == days }
+  end
+end
+
+# calculate for year of 2022 and months with 30 days
+puts NewDate.new(2022).get_months_with_days(30)
```
Comment (cyberik-dev) [inline 5 - months with days/solution_5.rb]: all text files should end with a newline

File: 5 - months with days/solution_5.rb (hunk 6 above)
Comment (neb5562) [reply 5 - months with days/solution_5.rb]: @cyberik-dev, Oleksi told us to move constants above class

File: 5 - months with days/solution_5.rb (hunk 7 above)
Comment (neb5562) [reply 5 - months with days/solution_5.rb]: @cyberik-dev  added parenthesis

File: 5 - months with days/solution_5.rb (hunk 6 above)
Comment (neb5562) [reply 5 - months with days/solution_5.rb]: @alexshvedko

Comment (cyberik-dev) [reply 5 - months with days/solution_5.rb]: https://www.rubyguides.com/2017/07/ruby-constants/

if you only use these constants inside a class, you must put them inside that class

File: 5 - months with days/solution_5.rb [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  def generate_months_with_days_count
+    months_with_days = {}
+    (FIRST_MONTH..LAST_MONTH).each do |i|
+      months_with_days[Date::MONTHNAMES[i].to_s] = Date.new(@year, i, NEGATIVE_NUM).day
+    end
+    months_with_days
+  end
+
+  # select months of year with 30 days or any value days variable gets.
+  def get_months_with_days(days)
```
Comment (cyberik-dev) [inline 5 - months with days/solution_5.rb]: please remove get_ from method names. I wrote about this in the chat
https://jtway.co/how-to-name-variables-and-methods-in-ruby-b4a21fb66c4d

File: 5 - months with days/solution_5.rb [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+FIRST_MONTH = 1
+LAST_MONTH = 12
+
+class NewDate
+  def initialize(year)
+    @year = year
+    @months_with_days = generate_months_with_days_count
+  end
+
+  def generate_months_with_days_count
```
Comment (cyberik-dev) [inline 5 - months with days/solution_5.rb:18]: alternative implementation:  

```ruby
def generate_months_with_days_count
    (FIRST_MONTH..LAST_MONTH).each.with_index(1).inject({}) do |hash, (_, i)| 
      month_name = Date::MONTHNAMES[i].to_s
      hash[month_name] = Date.new(@year, i, NEGATIVE_NUM).day
      hash
    end
  end
```

Comment (cyberik-dev) [inline 5 - months with days/solution_5.rb:18]: also this method should be in private section because you only use it inside class

**Review by cyberik-dev — CHANGES_REQUESTED** [review]

File: 5 - months with days/solution_5.rb (hunk 10 above)
Comment (cyberik-dev) [reply 5 - months with days/solution_5.rb:18]: but we can leave your solution because it is more readable

Comment (neb5562) [reply 5 - months with days/solution_5.rb:18]: @cyberik-dev we need to return `hash` in every `each` iteration?

Comment (neb5562) [reply 5 - months with days/solution_5.rb:18]: @cyberik-dev fixed

File: 5 - months with days/solution_5.rb (hunk 9 above)
Comment (neb5562) [reply 5 - months with days/solution_5.rb]: @cyberik-dev removed `get_`

File: 5 - months with days/solution_5.rb (hunk 6 above)
Comment (neb5562) [reply 5 - months with days/solution_5.rb]: @alexshvedko i get it now. moved those constants inside

File: 5 - months with days/solution_5.rb (hunk 8 above)
Comment (neb5562) [reply 5 - months with days/solution_5.rb]: @cyberik-dev i configured VS code to add new line in end of every file.

**Pull request merged**